### PR TITLE
Update documentation with plugins, related tools etc.

### PR DESCRIPTION
**Pull request by MattWoodhead**
An update to the documentation to add reference to some modules using the plugin interface, and in adition add some references to tools related to python-can to address #1041

Todo:
- [x]  Add example plugins to optional dependencies list as already noted elsewhere in the docs.
- [x]  Add more details and modules to the new "Other Tools" page
- [x]  Check documentation changes render correctly.

### CODE REVIEW

**Comment by zariiii9003** [comment]
@MattWoodhead I will review this on the weekend or next week. For now, could you resolve the new sphinx warnings?
https://github.com/hardbyte/python-can/actions/runs/3680103958/jobs/6225323383#step:5:118

File: doc/interfaces.rst [hunk 1]
```diff
@@ -39,3 +39,5 @@ The available hardware interfaces are:
    interfaces/usb2can
    interfaces/vector
 
+
+Additional interface types can be added via the :ref:`plugin interface`, or by installing a module that utilises the :ref:`plugin interface`.
```
Comment (hardbyte) [inline doc/interfaces.rst:43]: 👍🏼

File: doc/other-tools.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
+Other CAN bus tools
+===================
+
+In order to keep the project maintainable, the scope of the module is limited to providing common
```
Comment (hardbyte) [inline doc/other-tools.rst:4]:
```suggestion
In order to keep the project maintainable, the scope of the package is limited to providing common
```

File: doc/other-tools.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    .. _`CAN class`: https://docs.micropython.org/en/latest/library/pyb.CAN.html
+#. ASAM MDF Files
+    * The `asammdf`_ module provides many methods for processing ASAM (Association for
+      Standardization of Automation and Measuring Systems) MDF (Measurement Data Format) files.
+    .. _`asammdf`: https://asammdf.readthedocs.io/en/master/
+
+|
+|
+.. note::
+See also the available plugins for python-can in :ref:`plugin interface`.
```
Comment (hardbyte) [inline doc/other-tools.rst:75]:
```suggestion
   See also the available plugins for python-can in :ref:`plugin interface`.

```

**Review by hardbyte — APPROVED** [review]
Thanks, great job

File: setup.py [hunk 4]
```diff
@@ -30,13 +30,20 @@
...
     "gs_usb": ["gs_usb>=0.2.1"],
     "nixnet": ["nixnet>=0.3.1"],
     "pcan": ["uptime~=3.0.1"],
+    "remote": ["python-can-remote"],
+    "sontheim": ["python-can-sontheim>=0.1.2"],
     "viewer": [
         'windows-curses;platform_system=="Windows" and platform_python_implementation=="CPython"'
     ],
 }
+# Add a tag that allows the user to install all optional dependencies at once
```
Comment (zariiii9003) [inline setup.py:43]: I'm not sure about this. Once we migrate to pyproject.toml configuration this might be problematic. I'm not sure if there is syntax like this in PEP 621. This is also out of scope for this PR.

@hardbyte what do you think?

**Review by zariiii9003 — CHANGES_REQUESTED** [review]
Overall great work, thank you for the contribution. 👍

Some things should be changed though.

File: setup.py (hunk 4 above)
Comment (hardbyte) [reply setup.py:43]: Agree, out of scope for this PR.

Comment (MattWoodhead) [reply setup.py]: Extra tag removed :thumbsup:

**Comment by zariiii9003** [comment]
I just noticed, the enumerated lists do not seem to work as intended:
image

File: doc/other-tools.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      variables in an ECU.
+
+.. _pyxcp: https://pyxcp.readthedocs.io/en/latest/
+
+CAN Frame Parsing tools etc. (implemented in Python)
+----------------------------------------------------
+
+#. CAN Message / Database scripting
+    * The `cantools`_ package provides multiple methods for interacting with can message database
+      files, and using thjese files to monitor live busses with a command line monitor tool.
```
Comment (zariiii9003) [inline doc/other-tools.rst:57]:
```suggestion
      files, and using these files to monitor live busses with a command line monitor tool.
```

**Comment by MattWoodhead** [comment]
> I just noticed, the enumerated lists do not seem to work as intended:

Sorry, I think I must have broken those when I tried to sort out the sphinx warnings.

**Review by zariiii9003 — APPROVED** [review]

**Comment by felixdivo** [comment]
Awesome!
